pkg/builder/image.go

```diff
@@ -120,7 +120,7 @@ func jvmDockerfile(ctx *builderContext) error {
 }
 
 func incrementalImageContext(ctx *builderContext) error {
-	if ctx.Build.BaseImage != "" {
+	if ctx.Build.BaseImage != "" && ctx.Build.BaseImage != defaults.BaseImage() {
```

Comment:
True. I fail to understand this check meaning then and how it worked before the last refactoring. The `ctx.Build.BaseImage` will always be `!= ""` because, at minimum it is filled with the default base image, reason why it is not working. Any help appreciated :)

Reply:
The branch logic, to fallback to "standard" image, should be removed.

The logic to configure either "standard" or "incremental" should be done in the _builder_ trait, where a configuration parameter could be added, and the corresponding build step added. It's a bit more subtle than that, because "incremental" image build step should be deactivated for Quarkus native build, from the _quarkus_ trait, that already manipulates the build steps, but that should be quite straightforward.

There is also the following TODO that may be removed while we are at it:

https://github.com/apache/camel-k/blob/478ca56e54eadc3cfa9a10bf592935a34d37d9e7/pkg/builder/image.go#L148

Reply:
Okey, the goal of this PR is to fix the incremental build which is failing since version 1.7. I am going to remove that branch to fix in `1.8.1` and create a follow up issue to implement the suggestions you've provided.